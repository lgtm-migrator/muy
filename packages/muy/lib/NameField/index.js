import React from "react"
import TextField from "@material-ui/core/TextField"

const NameField = ({ id, inputProps, maxLength, name, required, ...props }) => (
  <TextField
    aria-required={required}
    id={id}
    name={name}
    inputProps={{
      "data-name": name,
      maxLength,
      ...inputProps,
    }}
    required={required}
    type={"text"}
    {...props}
  />
)

export default NameField

NameField.defaultProps = {
  id: "name",
  maxLength: 256,
  name: "name",
}
