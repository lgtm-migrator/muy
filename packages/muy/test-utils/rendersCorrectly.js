import renderer from "react-test-renderer"

const rendersCorrectly = component => {
  const tree = renderer.create(component).toJSON()
  return expect(tree).toMatchSnapshot()
}

export default rendersCorrectly
