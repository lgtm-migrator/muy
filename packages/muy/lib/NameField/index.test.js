import React from "react"
import rendersCorrectly from "../../test-utils/rendersCorrectly"
import NameField from "."

describe("<NameField /> component", () => {
  it("should render correctly with no props", () => {
    rendersCorrectly(<NameField />)
  })
  it("should render correctly when required", () => {
    rendersCorrectly(<NameField required={true} />)
  })
  it("should render correctly with no required", () => {
    rendersCorrectly(<NameField required={false} />)
  })
})
