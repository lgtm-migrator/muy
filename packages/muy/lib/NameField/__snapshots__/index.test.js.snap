// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NameField /> component should render correctly when required 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      aria-required={true}
      autoCapitalize="words"
      autoComplete="name"
      autoCorrect="off"
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      data-name="name"
      disabled={false}
      id="name"
      inputMode="latin-name"
      maxLength={50}
      minLength={5}
      name="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={true}
      size={50}
      tabIndex={0}
      type="text"
    />
  </div>
</div>
`;

exports[`<NameField /> component should render correctly with no props 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoCapitalize="words"
      autoComplete="name"
      autoCorrect="off"
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      data-name="name"
      disabled={false}
      id="name"
      inputMode="latin-name"
      maxLength={50}
      minLength={5}
      name="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      size={50}
      tabIndex={0}
      type="text"
    />
  </div>
</div>
`;

exports[`<NameField /> component should render correctly with no required 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      aria-required={false}
      autoCapitalize="words"
      autoComplete="name"
      autoCorrect="off"
      autoFocus={false}
      className="MuiInputBase-input MuiInput-input"
      data-name="name"
      disabled={false}
      id="name"
      inputMode="latin-name"
      maxLength={50}
      minLength={5}
      name="name"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      size={50}
      tabIndex={0}
      type="text"
    />
  </div>
</div>
`;
